lib/tapioca/rbi/printer.rb

```diff
@@ -0,0 +1,406 @@
+
+    class Const
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        previous_node = v.previous_node
+        v.printn if previous_node && (!previous_node.oneline? || !oneline?)
+
+        v.printl("#{name} = #{value}")
+      end
+    end
+
+    class Method
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        previous_node = v.previous_node
+        v.printn if previous_node && (!previous_node.oneline? || !oneline?)
+
+        v.visit_all(sigs)
+        v.printt
+        unless v.in_visibility_group || visibility == Visibility::Public
+          v.print(visibility.visibility.to_s)
+          v.print(" ")
+        end
+        v.print("def ")
+        v.print("self.") if is_singleton
+        v.print(name)
+        unless params.empty?
+          v.print("(")
+          params.each_with_index do |param, index|
+            v.print(", ") if index > 0
+            v.visit(param)
+          end
+          v.print(")")
+        end
+        v.print("; end")
+        v.printn
+      end
+
+      sig { override.returns(T::Boolean) }
+      def oneline?
+        sigs.empty?
+      end
+    end
+
+    class Param
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        v.print(name.to_s)
+      end
+    end
+
+    class OptParam
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        v.print("#{name} = #{value}")
+      end
+    end
+
+    class RestParam
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        v.print("*#{name}")
+      end
+    end
+
+    class KwParam
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        v.print("#{name}:")
+      end
+    end
+
+    class KwOptParam
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        v.print("#{name}: #{value}")
+      end
+    end
+
+    class KwRestParam
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        v.print("**#{name}")
+      end
+    end
+
+    class BlockParam
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        v.print("&#{name}")
+      end
+    end
+
+    class Mixin
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        case self
+        when Include
+          v.printt("include")
+        when Extend
+          v.printt("extend")
+        when MixesInClassMethods
+          v.printt("mixes_in_class_methods")
+        end
+        v.printn(" #{name}")
+      end
+    end
+
+    class Visibility
+      extend T::Sig
+
+      sig { override.params(v: Printer).void }
+      def accept_printer(v)
+        v.printl(visibility.to_s)
+      end
```

Comment:
Like @KaanOzkan said, it would definitely make the privacy level of the method more accessible, at the risk of making the RBI more verbose. I like the idea and we can have it as the default, but it would be nice to allow this to be configurable, if possible.

Reply:
I'll provide a follow up PR making this kind of option available from the CLI, wdyt?